### PR TITLE
Run GitHub Actions on changes to main branch

### PR DESCRIPTION
**Pull request by imjasonh**
I don't know when the default branch changed to `main`, but these checks haven't run since whenever that was.

cc @dlorenc

### CODE REVIEW

**Review by loosebazooka — APPROVED** [review]

**Comment by imjasonh** [comment]
> looks like you're still making changes

Yeah, there are some CI failures I'm trying (and mostly failing) to debug. Since Travis wasn't running for the last ~month, I fear real breakages might have crept in. 😬

**Comment by loosebazooka** [comment]
I vaguely remember @dlorenc  said he disabled a lot of the experimental builds to get it to build again

**Comment by imjasonh** [comment]
Okay, I got it working 😓 .

Instead of testing everything (some of which is secretly broken, yay!), CI as defined in this PR now only runs `bazel build //base/...`. We can build on this to include more targets, eventually building/testing everything with exceptions (`bazel test -- //... -//excluded:target`), until those exceptions are either fixed or deleted.

It sounds like dotnet and java8 are good candidates for delortation, others might be lurking, we can address them in future PRs.

**Review by loosebazooka — APPROVED** [review]

**Comment by chanseokoh** [comment]
I believe `bazel build //base/...` is almost empty, if not complete empty. I recall most of the actual tests are marked as "manual."

**Comment by imjasonh** [comment]
> I believe `bazel test //base/...` is almost empty, if not complete empty. I recall most of the actual tests are marked as "manual."

After removing `//experimental/dotnet`, we can now `bazel test //...` -- I'm not sure why java8 failed earlier and doesn't anymore. 🤷

**Comment by chanseokoh** [comment]
What I'm saying is, without `bazel test ... $(bazel query 'attr("tags", "'${GOARCH}'", "//...")')`, you are missing most of actual tests. [CONTRIBUTING.md](https://github.com/GoogleContainerTools/distroless/blob/main/CONTRIBUTING.md) says

> (`bazel test //...` will NOT run all tests, as many tests are marked "manual".)

**Comment by imjasonh** [comment]
@chanseokoh Thanks for clarifying, I've re-added all those targets, except for java8 and experimental, which fail for me.

**Comment by imjasonh** [comment]
Had to update some testdata to fix errors in java11 tests, but now it's all passing! 🎉

**Comment by imjasonh** [comment]
Friendly ping. Good to merge? 🙏

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -1,36 +1,32 @@
...
-        # Run all tests tagged with our architecture:
-        bazel test  --cpu=${CPU} --curses=no --test_output=errors --test_timeout=900 $(bazel query 'attr("tags", "'${GOARCH}'", "//...")')
-
+        set -ex
+        bazel build --curses=no //package_manager:dpkg_parser.par
+        bazel build --curses=no //...
+        bazel test --curses=no --test_output=errors //...
+        targets=$(bazel query 'attr("tags", "'amd64'", "//...")' | grep -v java8 | grep -v experimental)
+        bazel build --curses=no ${targets}
+        bazel test --curses=no --test_output=errors ${targets}
```
Comment (loosebazooka) [inline .github/workflows/ci.yaml:32]: can we add in a comment/TODO on what's missing now?

**Review by loosebazooka — COMMENTED** [review]
looks good, just want to make sure we know how to followup

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (imjasonh) [reply .github/workflows/ci.yaml:34]: Done.

**Review by loosebazooka — APPROVED** [review]

**Pull request merged**